odoo/addons/base/tests/test_qweb.py

```diff
@@ -1705,6 +1705,39 @@ def test_render_widget_duration_fallback(self):
         })
         self.env['ir.qweb'].with_context(lang='pt_BR')._render(view1.id, {})  # should not crash
 
+    def test_render_template_from_file(self):
+        expected_result = etree.fromstring(file_open('base/tests/file_template/file_expected_render.xml').read())
+        rendered_result = self.env['ir.qweb']._render('base/tests/file_template/templates/file_template.xml', values={
+            'document_name': 'Test Document',
+            'partner': {
+                'name': 'Jerry',
+                'forename': 'Khan',
+            },
+        })
+        self.assertEqual(etree.fromstring(rendered_result), expected_result)
+
+    def test_render_template_from_file_special_cases(self):
+        with self.assertRaises(ValueError):
+            self.env['ir.qweb']._render('base/tests/file_template/templates/../unreadable_file_template.xml', values={})
```

Comment:
Ok, got it. 

Still, for clarity, I would add comments explaining what we check in both the asertRaises, and would put this one down below, with the other one.